### PR TITLE
DSC19-21: Setup and configure HAPI FHIR

### PR DESCRIPTION
**Pull request by de-laz**
This PR can be tested by running the Instant OpenHIE with the following command:
`yarn docker:instant init -t docker covid19surveillance -c="<path WHO COVID-19 Surveillance Package>"`
To test you can send a `GET` request to http://localhost:3447/fhir/ImplementationGuide

### CODE REVIEW

File: docker/importer/docker-compose.config.yml [hunk 1]
```diff
@@ -34,6 +34,21 @@ services:
         target: /instant
     # This command will only attempt to import the covid19-surveillance-mediator config when the uptime endpoint responds with 200
     command: sh -c "wait-on -t 60000 http-get://covid19-surveillance-mediator:3003/uptime && sleep 1 && node /instant/who-covid19-surveillance-package/docker/importer/volume/endpoint.js"
+  
+  hapi-fhir-config-importer:
+    container_name: hapi-fhir-config-importer
+    image: jembi/instantohie-config-importer
+    restart: on-failure
+    environment:
+      HAPI_FHIR_PATH: 'hapi-fhir/hapi-fhir-jpaserver/fhir/'
```
Comment (MattyJ007) [inline docker/importer/docker-compose.config.yml]: @de-laz Won't this `hapi-fhir-jpaserver` section be removed in the server version bump?

File: docker/importer/volume/fhirConfig.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  })
+}
+
+downloadResources((err, files) => {
+  if (err) {
+    console.error(err)
+  } else {
+    postToFHIRServer(files)
+  }
+})
```
Comment (MattyJ007) [inline docker/importer/volume/fhirConfig.js]: @de-laz Missing newline character

File: docker/importer/volume/fhirConfig.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+const fs = require('fs')
+const https = require('https')
+const http = require('http')
+const path = require('path')
+const { spawn } = require('child_process')
+
+const resourcesPath = path.resolve(__dirname, 'fhirResources')
+
+const downloadResources = (cbk) => {
```
Comment (MattyJ007) [inline docker/importer/volume/fhirConfig.js]: @de-laz What is `cbk`?

File: docker/importer/volume/fhirConfig.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  const COVID19_IG_URL = process.env.COVID19_IG_URL || 'https://openhie.github.io/covid-ig'
+  
+  if (fs.existsSync())
+    fs.rmdirSync(resourcesPath)
+  
+  fs.mkdirSync(resourcesPath)
+  
+  const resources = fs.createWriteStream(path.resolve(resourcesPath, 'fhir-resources.zip'))
```
Comment (MattyJ007) [inline docker/importer/volume/fhirConfig.js]: Why do we need the data written into files? You just read the files in the next step? Why not just hold them in memory.

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
Got some questions

File: docker/importer/volume/fhirConfig.js (hunk 3 above)
Comment (de-laz) [reply docker/importer/volume/fhirConfig.js]: @MattyJ007 The correct standard name used is usually `cb` for a `callback` function. I could revert this to just `callback` for clarity.

Comment (de-laz) [reply docker/importer/volume/fhirConfig.js]: @MattyJ007  Alternatively, we can turn the `downloadResources` function to a promise-returning function.

**Comment by MattyJ007** [comment]
> This PR can be tested by running the Instant OpenHIE with the following command:
> `yarn docker:instant init -t docker -c="<path WHO COVID-19 Surveillance Package>"`
> To test you can send a `GET` request to http://localhost:3447/fhir/ImplementationGuide


@de-laz This command is missing the IDs of the packages to start

Perhaps?
```
yarn docker:instant init covid19surveillance -c="../who-covid19-surveillance-package"
```

File: .prettierrc.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+semi: false
+singleQuote: true
+bracketSpacing: true
+trailingComma: "none"
+arrowParens: "avoid"
```
Comment (bradsawadye) [inline .prettierrc.yml]:
```suggestion
arrowParens: "avoid"

```

**Comment by bradsawadye** [comment]
@de-laz I'm getting 404s on the importer

**Pull request merged**